atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -380,6 +382,21 @@ private enum State {
         this.transactionOutcomeMetrics =
                 TransactionOutcomeMetrics.create(transactionMetrics, metricsManager.getTaggedRegistry());
         this.expectationsDataCollectionMetrics = ExpectationsMetrics.of(metricsManager.getTaggedRegistry());
+
+        // TODO (jkong): Fix path type tracking for tests:
+        this.keyValueSnapshotReader = new DefaultKeyValueSnapshotReader(
```

Comment:
Thoughts on actually this being given in the constructor? My master plan is actually that you'd replace this codepath with the snapshot reader (or at least rejig it a lot): https://github.com/palantir/atlasdb/blob/develop/atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransaction.java#L915.

This codepath is soo convoluted, very carefully crafting codepaths to turn off in SnapshotTransaction.

Reply:
yeah that's fine, the goal on my end was eventually to give you access to decide how this is created in your TransactionManager (or, following some internal discussions, something wrapping this)!